File: images-basics/IO.py
#-----------------------------------------------------------------------------
#
# Example for images input/output in Python3.
#
#-----------------------------------------------------------------------------

import cv2

# Read image with standard colors.
img = cv2.imread("../img.jpg")

# Read image with grayscale colors.
imgGrayscale = cv2.imread("../img.jpg", cv2.IMREAD_GRAYSCALE)

# Save image.
cv2.imwrite("img.jpg", img)

# For byte-IO:
#     cv2.imdecode()
#     cv2.imencode()

File: basics/IO/IO.cpp
//-----------------------------------------------------------------------------
//
// Example for images input/output in C++ with OpenCV.
//
//-----------------------------------------------------------------------------

#include <opencv2/opencv.hpp>

int main() {
    /* Read image with standard colors. */
    cv::Mat img = cv::imread("../img.jpg");

    /* Read image with Gray colors. */
    cv::Mat imgGrayscale = cv::imread("../img.jpg", cv::IMREAD_GRAYSCALE);

    /* Save image. */
    cv::imwrite("img.jpg", img);

    /*  For byte-IO:
     *      cv::imdecode()
     *      cv::imencode()
     */
}
